src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsClient/DiagnosticsClient.cs

```diff
@@ -30,6 +30,11 @@ public sealed class DiagnosticsClient
         {
         }
 
+        public DiagnosticsClient(string diagnosticPort) :
```

Comment:
```suggestion
        internal DiagnosticsClient(string diagnosticPort) :
```

Reply:
I would also add a comment warning not to make this public in the future without considering how TcpSocket will be handled.

Reply:
Removed the string constructor altogether with Noah's advice.